### PR TITLE
25 and other issues solve

### PR DESCRIPTION
**Pull request by IlyaMoskva**
Issue #25, #26, #27  and others.
Cumulative patch:
* Graph API implementation up to workable stage, common access to Graph as a Singleton.
* Core interfaces implementations to make system runnable and testable: Address, Network, Storage.
* Tests to confirm it works.

### CODE REVIEW

File: src/main/com/cqfn/degitx/simulator/Request.kt [hunk 1]
```diff
@@ -2,12 +2,14 @@ package com.cqfn.degitx.simulator
 
 /**
  * Stub for Request sent via Network to Address.
+ * Data class.
```
Comment (g4s8) [inline src/main/com/cqfn/degitx/simulator/Request.kt:5]: @IlyaMoskva you can use `data class` for data classes: https://www.geeksforgeeks.org/kotlin-data-classes/

**Review by g4s8 — CHANGES_REQUESTED** [review]
@IlyaMoskva please check review comments

File: src/main/com/cqfn/degitx/simulator/Address.kt [hunk 2]
```diff
@@ -1,9 +1,24 @@
...
  * Representation of address for emulator
  */
 class Address {
+
+    private val source = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+    private val addressLength = 6L
+
+    constructor() {
+        this.addr = Random().ints(addressLength, 0, source.length)
```
[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/Address.kt:16]: @g4s8 , this call is for test purpose mostly. Prime constructor accepts Address as a parameter. But anyway, this part will be modified as you suggested. Thanks.

[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/Address.kt:16]: Now the code is fixed but not brilliant, as static objects in kotlin are not as native as in java.

File: src/main/com/cqfn/degitx/simulator/BasicNetwork.kt [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.cqfn.degitx.simulator
+
+import java.util.*
+
+/**
+ * Network implementation as a very common network adapter.
```
[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/BasicNetwork.kt:6]: Fixed.

File: src/main/com/cqfn/degitx/simulator/DsGraph.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
+package com.cqfn.degitx.simulator
+
+import java.util.*
+
+object DsGraph : Graph {
```
[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/DsGraph.kt:5]: Added.

File: src/main/com/cqfn/degitx/simulator/Hardware.kt [hunk 5]
```diff
@@ -7,10 +7,10 @@ interface Hardware {
...
      * Data storage hardware
      */
-    val storage: Storage
+    var storage: Storage
 
     /**
      * Network adapter
      */
-    val net: Network
+    var net: Network
```
[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/Hardware.kt:15]: Hm.. Mistake, for sure. Fixed.

File: src/main/com/cqfn/degitx/simulator/NodeHardware.kt [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package com.cqfn.degitx.simulator
+
+class NodeHardware (override var storage: Storage,
```
[the remark this answers is not in the document]
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/NodeHardware.kt:3]: Done.

File: src/main/com/cqfn/degitx/simulator/Request.kt (hunk 1 above)
Comment (IlyaMoskva) [reply src/main/com/cqfn/degitx/simulator/Request.kt:5]: @g4s8 , you're right, there will not be any logic. Fixed.

**Comment by IlyaMoskva** [comment]
@g4s8 ,please re-review it.

**Review by g4s8 — APPROVED** [review]
@IlyaMoskva thanks

**Pull request merged**